noxfile.py

```diff
@@ -0,0 +1,173 @@
+######################################################################
+#
+# File: noxfile.py
+#
+# Copyright 2020 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+import os
+import subprocess
+
+import nox
+
+CI = os.environ.get('CI') is not None
+NOX_PYTHONS = os.environ.get('NOX_PYTHONS')
+
+PYTHON_VERSIONS = ['3.5', '3.6', '3.7', '3.8'] if NOX_PYTHONS is None else NOX_PYTHONS.split(',')
+PYTHON_DEFAULT_VERSION = PYTHON_VERSIONS[-1]
+
+PY_PATHS = ['b2', 'test', 'noxfile.py', 'setup.py']
+
+REQUIREMENTS_FORMAT = ['yapf==0.27']
+REQUIREMENTS_LINT = ['yapf==0.27', 'pyflakes', 'flake8==3.8.3', 'pytest==5.4.3']
+REQUIREMENTS_TEST = ['nose==1.3.7', 'pytest==5.4.3', 'pytest-cov==2.10.0']
+REQUIREMENTS_BUILD = ['liccheck==0.4.7', 'setuptools>=20.2']
+REQUIREMENTS_DOC = [
+    'sphinx', 'sphinx-autobuild', 'sphinx_rtd_theme', 'sphinx-argparse', 'sphinxcontrib-plantuml',
+    'sadisplay'
+]
+
+nox.options.reuse_existing_virtualenvs = True
+nox.options.sessions = [
+    'lint',
+    'test',
+]
+
+# In CI, use Python interpreter provided by GitHub Actions
+if CI:
+    nox.options.force_venv_backend = 'none'
+
+
+def install_myself(session):
+    """Install from the source."""
+    # In CI, install B2 SDK from the master branch
+    if CI:
+        session.install('git+git://github.com/Backblaze/b2-sdk-python#egg=b2sdk')
+
+    session.install('-e', '.')
+
+
+# noinspection PyShadowingBuiltins
+@nox.session(python=PYTHON_DEFAULT_VERSION)
+def format(session):
+    """Format the code."""
+    session.install(*REQUIREMENTS_FORMAT)
+    # TODO: incremental mode for yapf
+    session.run('yapf', '--in-place', '--parallel', '--recursive', *PY_PATHS)
+    # TODO: uncomment if we want to use isort and docformatter
+    # session.run('isort', *PY_PATHS)
+    # session.run(
+    #     'docformatter',
+    #     '--in-place',
+    #     '--recursive',
+    #     '--wrap-summaries=100',
+    #     '--wrap-descriptions=100',
+    #     *PY_PATHS,
+    # )
+
+
+@nox.session(python=PYTHON_DEFAULT_VERSION)
+def lint(session):
+    """Run linters."""
+    session.install(*REQUIREMENTS_LINT)
+    session.run('yapf', '--diff', '--parallel', '--recursive', *PY_PATHS)
+    # TODO: uncomment if we want to use isort and docformatter
+    # session.run('isort', '--check', *PY_PATHS)
+    # session.run(
+    #     'docformatter',
+    #     '--check',
+    #     '--recursive',
+    #     '--wrap-summaries=100',
+    #     '--wrap-descriptions=100',
```

Comment:
can you do something about the clones please?

Reply:
What do you mean?

Reply:
we have identical here and there code except for `--check`. Is there no better way?

Reply:
IMO, there is no more readable way :) Unfortunately, docformatter doesn't support any configuration file so I could remove `--wrap-summaries` and `--wrap-descriptions`